File: src/main/java/com/kingmang/lazurite/console/output/impl/SystemOutput.java
package com.kingmang.lazurite.console.output.impl;

import com.kingmang.lazurite.console.output.Output;

import java.io.File;

public class SystemOutput implements Output {

    private final PrintStreamOutput out = new PrintStreamOutput(System.out, System.err);

    @Override
    public String newline() {
        return System.lineSeparator();
    }

    @Override
    public void print(String value) {
        out.print(value);
    }

    @Override
    public void println(String value) {
        out.println(value);
    }

    @Override
    public String getText() {
        return "";
    }

    @Override
    public void error(CharSequence value) {
        out.error(value);
    }

    @Override
    public File fileInstance(String path) {
        return new File(path);
    }
}

File: src/main/java/com/kingmang/lazurite/console/output/impl/SystemOutput.kt
package com.kingmang.lazurite.console.output.impl

import com.kingmang.lazurite.console.output.Output
import java.io.File

class SystemOutput : Output {
    private val out = PrintStreamOutput(System.out, System.err)

    override fun newline(): String =
        System.lineSeparator()

    override fun print(value: String) {
        out.print(value)
    }

    override fun error(value: CharSequence) {
        out.error(value)
    }

    override fun fileInstance(path: String): File =
        File(path)
}
